File: sdk/go/sdk/buildkite/wait_step.go
package buildkite

import "github.com/buildkite/buildkite-sdk/sdk/go/sdk/schema"

type WaitStep struct {
	AllowDependencyFailure *bool
	Branches               []string
	ContinueOnFailure      *bool
	DependsOn              DependsOn
	ID                     *string
	Identifier             *string
	If                     *string
	Key                    *string
	Label                  *string
	Name                   *string
	Wait                   *string
}

func (step WaitStep) toPipelineStep() *PipelineStep {
	waitStep := &PipelineStep{
		ID:         step.ID,
		Identifier: step.Identifier,
		If:         step.If,
		Key:        step.Key,
		Label:      step.Label,
		Name:       step.Name,
		Wait: &schema.Label{
			String: step.Wait,
		},
	}

	if step.AllowDependencyFailure != nil {
		waitStep.AllowDependencyFailure = &schema.AllowDependencyFailureUnion{
			Bool: step.AllowDependencyFailure,
		}
	}

	if step.Branches != nil {
		waitStep.Branches = &schema.Branches{
			StringArray: step.Branches,
		}
	}

	if step.ContinueOnFailure != nil {
		waitStep.ContinueOnFailure = &schema.AllowDependencyFailureUnion{
			Bool: step.ContinueOnFailure,
		}
	}

	if step.DependsOn != nil {
		waitStep.DependsOn = step.DependsOn.toSchema()
	}

	return waitStep
}

File: sdk/csharp/src/Buildkite.Sdk/Schema/WaitStep.cs
using System.Text.Json.Serialization;
using YamlDotNet.Serialization;

namespace Buildkite.Sdk.Schema;

/// <summary>
/// A wait step waits for all previous steps to complete before continuing.
/// </summary>
public class WaitStep : IStep, IGroupStep
{
    /// <summary>Optional label for the wait step.</summary>
    public string? Wait { get; set; } = "";

    /// <summary>Alias for Wait.</summary>
    [JsonIgnore]
    [YamlIgnore]
    public string? Label { get => Wait; set => Wait = value; }

    /// <summary>Alias for Wait.</summary>
    [JsonIgnore]
    [YamlIgnore]
    public string? Name { get => Wait; set => Wait = value; }

    /// <summary>A unique identifier for this step.</summary>
    public string? Key { get; set; }

    /// <summary>Alias for Key.</summary>
    [JsonIgnore]
    [YamlIgnore]
    [Obsolete("Use Key instead.")]
    public string? Id { get => Key; set => Key = value; }

    /// <summary>Alias for Key.</summary>
    [JsonIgnore]
    [YamlIgnore]
    public string? Identifier { get => Key; set => Key = value; }

    /// <summary>A boolean expression to conditionally run this step.</summary>
    public string? If { get; set; }

    /// <summary>Branch filter pattern.</summary>
    public object? Branches { get; set; }

    /// <summary>Step keys this step depends on.</summary>
    public DependsOn? DependsOn { get; set; }

    /// <summary>Whether to proceed if a dependency fails.</summary>
    public bool? AllowDependencyFailure { get; set; }

    /// <summary>Continue even if previous steps failed.</summary>
    public bool? ContinueOnFailure { get; set; }
}
